extensions/shared/components/audio-player/index.jsx

```diff
@@ -0,0 +1,153 @@
+/* global _wpmejsSettings, MediaElementPlayer */
+
+/**
+ * External dependencies
+ */
+import { throttle } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { speak } from '@wordpress/a11y';
+
+/**
+ * Internal dependencies
+ */
+import { STATE_PLAYING, STATE_PAUSED, STATE_ERROR } from './constants';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const meJsSettings = typeof _wpmejsSettings !== 'undefined' ? _wpmejsSettings : {};
+
+function createJumpButton( containerClass, label, clickHandler ) {
+	const buttonContainer = document.createElement( 'div' );
+	buttonContainer.className = containerClass;
+
+	const button = document.createElement( 'button' );
+	button.innerText = label;
+	button.addEventListener( 'click', clickHandler );
+	buttonContainer.appendChild( button );
+	return buttonContainer;
+}
+
+function AudioPlayer( {
+	trackSource,
+	onPlay,
+	onPause,
+	onError,
+	onTimeChange,
+	onSkipForward,
+	onJumpBack,
+	currentTime,
+	playStatus = STATE_PAUSED,
+} ) {
+	const audioRef = useRef();
+
+	/**
+	 * Play current audio.
+	 *
+	 * @public
+	 */
+	const play = () => {
+		// Ignoring exceptions as they are handled globally from the audio element.
+		audioRef.current.play().catch( () => {} );
+	};
+
+	/**
+	 * Pause current audio.
+	 *
+	 * @public
+	 */
+	const pause = () => {
+		audioRef.current.pause();
+		speak( __( 'Paused', 'jetpack' ), 'assertive' );
+	};
+
+	useEffect( () => {
+		const audio = audioRef.current;
+		// Initialize MediaElement.js.
+		const mediaElement = new MediaElementPlayer( audio, meJsSettings );
+
+		// Add the skip and jump buttons if needed
+		if ( onJumpBack || onSkipForward ) {
```

Comment:
Ideally, I'd like these components to be children of the audio player and for this code to call `addControlElement` for each of them. That way we can compose an audio player with whatever controls we like.

I spent a while trying to get this to work, but the code got quite messy so I decided to keep it as two props and handled within the component for now.

Reply:
Personally, I'm not totally convinced about this part. Looks a little bit confusing. The properties for callback functions don't define the availability of the controls in the IU. It means I can create an AudioPlayer instance...

```es6
<AudioPlayer />
```

... and even so, it will show the play control there. I guess we should mimic a similar behavior for the forward and back buttons, maybe by adding `controls` property following the [mediaelement API](https://github.com/mediaelement/mediaelement/blob/master/docs/api.md#attributes).

> controls | Specifies that video controls should be displayed (such as a play/pause button etc).
> -- | --

I imagine something like...

```es6
<AudioPlayer
    controls=[ 'play', 'pause', 'back', 'forward' ]
/>
```

Take this as a suggestion 👍

Reply:
Yeah that's what I'd like to do. I saw the controls as being child components of the `AudioPlayer` but it's really hard to achieve with the crossing between React and the MediaElement API.

We could create the forward and back controls as MediaElement plugins and then provide them as you suggest, but I'd like there to be the ability to set other props like callbacks, the amount of time to jump, etc.

Reply:
I think it would be good to solve this when we make the more composable version of this component.